### DISCUSSION TITLE
My Github Contributions keeps Reducing.

### DISCUSSION THREAD

**Discussion by Vikrez22** [community/community#130548] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Why do my GitHub contribution keep reducing, I can calculate over 1k+ contribution I have made from the year I started, but my contribution has refused to reach 1k, once it gets to 950-970 it comes down to 930-940.

I have noticed this for a while now, and it is not nice.
Please can anyone with a similar situation help out?

Thanks in advance.

**Comment by asdacosta** [comment]
There can be many reasons. This will help you identify the actual cause:
https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/managing-contribution-settings-on-your-profile/why-are-my-contributions-not-showing-up-on-my-profile

**Comment by AmirSarrafzadeh** [comment]
These are some of the reasons of reducing Github contibutions:

Rolling Window: GitHub contributions are shown on a rolling 365-day basis. This means contributions made more than a year ago will drop off the contribution graph as time progresses.

Types of Contributions Counted:

Commits in the default branch or the gh-pages branch.
Pull requests opened and merged.
Issues opened.
Contribution Visibility:

Contributions to private repositories are only visible to you, but they still count towards your overall contribution count.
Ensure that your contributions are being made to the correct branches and repositories that are counted towards your contributions.
Merge Activity:

If your contributions include merges or you are part of a repository where your changes get squashed, some of your individual commits might not be counted.
Local Commits:

Ensure that all your local commits are pushed to GitHub. Commits that are not pushed to GitHub won't be counted.
Time Zone Differences:

Contributions can be affected by time zone differences, which may cause discrepancies in contribution counting.
Repository Visibility Changes:

If a repository changes from public to private, the contributions in that repository will no longer be counted publicly.
Filtering Contributions:

Sometimes filtering settings in your contributions graph might hide some contributions. Check the settings on your contributions graph.

**Reply by ebndev** [reply, marked as answer]
Hi @AmirSarrafzadeh, thanks for taking the time to reply!

@Vikrez22 it's correct that your contribution graph is reflective of all of the days currently being displayed, so the number of contributions as well as the contribution colors may change. Please see this [related discussion](https://github.com/orgs/community/discussions/23261#discussioncomment-3239758) for more context and let us know if you have any additional questions 😄

**Reply by Vikrez22** [reply]
Thanks for the reply, Understood.

**Reply by Kimbohlovette** [reply]
For example, today, my contributions for this year are 783 according to my GitHub profile. When the number reaches a specific value, like 800, it drops back to 742. This cycle repeats. I can never go beyond a certain number of contributions. I've never changed my email like in the other cases.

Please, I need help with this. I've seen many related issues, but it seems they all result from changing or removing an email from an account. However, mine is different. I have never changed my email since I created my account.

**Comment by ThinkOutSideTheBlock** [comment]
I have the same problem as well, I don't know why my overall contributions go down and down.

**Reply by ebndev** [reply]
Hi @ThinkOutSideTheBlock!

Please see my comment above and that might help explain why your contribution number changes!
>  it's correct that your contribution graph is reflective of all of the days currently being displayed, so the number of contributions as well as the contribution colors may change. Please see this [related discussion](https://github.com/orgs/community/discussions/23261#discussioncomment-3239758) for more context and let us know if you have any additional questions 😄

**Reply by ThinkOutSideTheBlock** [reply]
hi @ebndev  ,Tnx for your reply,Problem is not only my total contributes get lower and lower but also when ever I contribute to other users repos , that doesn't count for me any more , so in last few days I have some commits in my on going projects on different repos and different branches but none of has been counted or shown in graph .

SO I think these are two separate problems but some how related to each other

**Reply by Kimbohlovette** [reply]
My contributions keep dropping to 920 - 930 each time I reach close to 1k

**Comment by MeydhiXploit** [comment]
i think your connection

**Comment by 0xJonaseb11** [comment]
I have the same problem! I had over 2,000 contributions overall but today I checked and found having 1,863! I really don't know why. Maybe the counting algorithms of github are going down and down, with time!!
Please help us get to fully understand why and also offer help with our contribution as developers if possible!
Thanks,
Jonas!!

**Reply by Kimbohlovette** [reply]
It's so annoying @0xJonaseb11. Suffered that the whole of last year and this year is not any different

**Reply by 0xJonaseb11** [reply]
> It's so annoying @0xJonaseb11. Suffered that the whole of last year and this year is not any different

yeah, man!! I think it is time to rethink about this and / or move to other technologies like oracle, gitlab etc - if such issues persist!!

**Comment by Kimbohlovette** [comment]
@ebndev  Please there something a bit more than just the contribution graph  in this issue. I've witness this issue where  my contributions add up to a certain number and then drop.  This has been happening to me over the years but the most recent one was last week. I had 1145 contributions but the following day it dropped to 1104. This is so annoying

**Comment by Vikrez22** [comment]
I figured out what’s actually happening.

GitHub’s contribution total is a rolling 12-month window, not a lifetime counter. As each new day passes, GitHub removes the contributions you made on the same date last year while adding today’s contributions.

So if you were very active around this time last year and your current daily activity is lower (or equal), your total can appear to drop or stagnate, even though you’re still contributing.

Nothing is being deleted. If you check your year-by-year contribution breakdown, you’ll see that all your contributions are still fully counted there.

It’s confusing, but this behavior is expected based on how GitHub calculates the contribution graph.
